power: Fix naming of Kconfig options controlling low power states

pizi-nordic opened the pull request:
The SYS_POWER_LOW_POWER_STATE_SUPPORTED and SYS_POWER_LOW_POWER_STATE
suggests one low power state but these options control multiple
low power state. This commit uses plural in the names to indicate
that.

pizi-nordic committed:
power: Fix naming of Kconfig options controlling low power states

The SYS_POWER_LOW_POWER_STATE_SUPPORTED and SYS_POWER_LOW_POWER_STATE
suggests one low power state but these options control multiple
low power state. This commit uses plural in the names to indicate
that.

Signed-off-by: Piotr Zięcik <piotr.ziecik@nordicsemi.no>

pizi-nordic commented:
recheck

pizi-nordic commented:
recheck

mnkp commented on kernel/Kconfig.power_mgmt:
```diff
@@ -49,7 +49,7 @@ config SYS_POWER_LOW_POWER_STATE
 	  saving most power.
 
 config SYS_POWER_DEEP_SLEEP
```
Since we are reworking names I think we should use the same convention for both low power and deep sleep. I.e. we should choose one of the pairs:
    * `SYS_POWER_LOW_POWER`, `SYS_POWER_DEEP_SLEEP`
    * `SYS_POWER_LOW_POWER_STATES`, `SYS_POWER_DEEP_SLEEP_STATES`

The former seems more general, indicates that we are enabling low power support. It may be a better choice. Especially that many SoCs enable now `SYS_POWER_LOW_POWER_STATES_SUPPORTED` even if they don't provide yet any support for low power mode, i.e. no implementation of `sys_set_power_state()` function.

pizi-nordic replied:
Fixed.

mnkp commented on kernel/Kconfig.power_mgmt:
```diff
@@ -49,7 +49,7 @@ config SYS_POWER_LOW_POWER_STATE
 
 config SYS_POWER_DEEP_SLEEP
-	bool "Deep sleep state"
+	bool "Enable Deep Sleep states"
```
"Enable" in Kconfig option is implied, i.e. all the options are there to be enabled. There is no need to be explicit or otherwise all the Kconfig options would start with "Enable ...". It's better to have
```
	bool "Deep Sleep states"
```

pizi-nordic replied:
Fixed

pizi-nordic committed:
power: Fix naming of Kconfig options controlling deep sleep states

This commit changes the names of SYS_POWER_DEEP_SLEEP* Kconfig
options in order to match SYS_POWER_LOW_POWER_STATE* naming
scheme.

Signed-off-by: Piotr Zięcik <piotr.ziecik@nordicsemi.no>

carlescufi reviewed (approved):
LGTM
